packages/Search/src/DTK_DistributedSearchTree_def.hpp

```diff
@@ -45,6 +45,19 @@ DistributedSearchTree<DeviceType>::DistributedSearchTree(
     _distributed_tree = std::make_shared<BVH<DeviceType>>( boxes );
 }
 
+template <typename DeviceType>
+typename DistributedSearchTree<DeviceType>::SizeType
+DistributedSearchTree<DeviceType>::size() const
+{
+    std::lock_guard<std::mutex> guard( cache_mutex );
```

Comment:
Don't you want to put the lock inside the `if`? Now every time the function is called, there is a lock. This seems overkilled.

Reply:
Putting it inside `if` would lead to race conditions, right?

Reply:
You may do the `allReduce` multiple times but you will always get the same result. The problem right now is that the `if` could be evaluated when `_size` is in an intermediary stage in the `allReduce`, i.e., the value is not `max` anymore but not yet the final value. With the mutex in the `if` you don't have this problem.

Reply:
Oh, so you are OK that if multiple threads enter `size` simultaneously, they will all do `allReduce` even if that's unnecessary?

Reply:
Yes, exactly. That's extra work but you still get the correct result.